u_values[u_sorter].searchsort would cause "Function path is chained attributes and name" but np.searchsort would not

The error-free version is:

```
import numpy as np

#pythran export _pythran_cdf_distance(int, float[:], float[:])
#pythran export _pythran_cdf_distance(int, float[:], float[:], None, None)
#pythran export _pythran_cdf_distance(int, float[:], float[:], float[:], float[:])
def _pythran_cdf_distance(p, u_values, v_values, u_weights=None, v_weights=None):
    u_sorter = np.argsort(u_values)
    v_sorter = np.argsort(v_values)

    all_values = np.concatenate((u_values, v_values))
    all_values = np.sort(all_values, kind='mergesort')

    # Compute the differences between pairs of successive values of u and v.
    deltas = np.diff(all_values)

    # Get the respective positions of the values of u and v among the values of
    # both distributions.
    u_cdf_indices = np.searchsorted(u_values[u_sorter],all_values[:-1],'right')
    v_cdf_indices = np.searchsorted(v_values[v_sorter],all_values[:-1],'right')

    # Calculate the CDFs of u and v using their weights, if specified.
    if u_weights is None:
        u_cdf = u_cdf_indices / u_values.size
    else:
        u_sorted_cumweights = np.concatenate(([0],
                                              np.cumsum(u_weights[u_sorter])))
        u_cdf = u_sorted_cumweights[u_cdf_indices] / u_sorted_cumweights[-1]

    if v_weights is None:
        v_cdf = v_cdf_indices / v_values.size
    else:
        v_sorted_cumweights = np.concatenate(([0],
                                              np.cumsum(v_weights[v_sorter])))
        v_cdf = v_sorted_cumweights[v_cdf_indices] / v_sorted_cumweights[-1]

    # Compute the value of the integral based on the CDFs.
    # If p = 1 or p = 2, we avoid using np.power, which introduces an overhead
    # of about 15%.
    if p == 1:
        return np.sum(np.multiply(np.abs(u_cdf - v_cdf), deltas))
    if p == 2:
        return np.sqrt(np.sum(np.multiply(np.square(u_cdf - v_cdf), deltas)))
    return np.power(np.sum(np.multiply(np.power(np.abs(u_cdf - v_cdf), p),
                                       deltas)), 1/p)
```

If I modify `u_cdf_indices = np.searchsorted(u_values[u_sorter],all_values[:-1],'right')` to `u_cdf_indices = u_values[u_sorter].searchsorted(all_values[:-1], 'right')`, it would cause `AssertionError: Function path is chained attributes and name`:

```
Traceback (most recent call last):
  File "/Users/charlotte/anaconda3/envs/scipydev/bin/pythran", line 11, in <module>
    sys.exit(run())
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/run.py", line 175, in run
    pythran.compile_pythranfile(args.input_file,
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/toolchain.py", line 472, in compile_pythranfile
    output_file = compile_pythrancode(module_name, open(file_path).read(),
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/toolchain.py", line 394, in compile_pythrancode
    module, error_checker = generate_cxx(module_name, pythrancode, specs, opts,
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/toolchain.py", line 138, in generate_cxx
    pm, ir, docstrings = front_middle_end(module_name, code, optimizations,
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/toolchain.py", line 105, in front_middle_end
    refine(pm, ir, optimizations)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/middlend.py", line 29, in refine
    pm.apply(NormalizeMethodCalls, node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/passmanager.py", line 236, in apply
    return a.apply(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/passmanager.py", line 199, in apply
    new_node = self.run(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/passmanager.py", line 187, in run
    n = super(Transformation, self).run(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/passmanager.py", line 96, in run
    return self.visit(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/passmanager.py", line 78, in visit
    return super(ContextManager, self).visit(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/ast.py", line 371, in visit
    return visitor(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/transformations/normalize_method_calls.py", line 48, in visit_Module
    self.generic_visit(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/ast.py", line 447, in generic_visit
    value = self.visit(value)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/passmanager.py", line 78, in visit
    return super(ContextManager, self).visit(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/ast.py", line 371, in visit
    return visitor(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/transformations/normalize_method_calls.py", line 63, in visit_FunctionDef
    self.generic_visit(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/ast.py", line 447, in generic_visit
    value = self.visit(value)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/passmanager.py", line 78, in visit
    return super(ContextManager, self).visit(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/ast.py", line 371, in visit
    return visitor(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/transformations/normalize_method_calls.py", line 83, in visit_Assign
    n = self.generic_visit(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/ast.py", line 456, in generic_visit
    new_node = self.visit(old_value)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/passmanager.py", line 78, in visit
    return super(ContextManager, self).visit(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/ast.py", line 371, in visit
    return visitor(node)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/transformations/normalize_method_calls.py", line 224, in visit_Call
    node.func.value, _ = rec(node.func.value, MODULES)
  File "/Users/charlotte/anaconda3/envs/scipydev/lib/python3.8/site-packages/pythran/transformations/normalize_method_calls.py", line 208, in rec
    assert isinstance(path, (ast.Name, ast.Attribute)), err
AssertionError: Function path is chained attributes and name
```

#### Reference
https://github.com/serge-sans-paille/pythran/issues/1064: This issue implements a similar function as mine, and I got the idea to rewrite `u_cdf_indices = u_values[u_sorter].searchsorted(all_values[:-1], 'right')` to `u_cdf_indices = np.searchsorted(u_values[u_sorter],all_values[:-1],'right')` to through comparing this issue's implementations with mine when debugging.
https://github.com/serge-sans-paille/pythran/issues/1791: This is a similar issue I opened. The function is the same.

--- a/pythran/tables.py
+++ b/pythran/tables.py
@@ -3979,7 +3979,7 @@ MODULES = {
         "rot90": ConstFunctionIntr(),
         "round": ConstMethodIntr(),
         "round_": ConstMethodIntr(),
-        "searchsorted": ConstFunctionIntr(),
+        "searchsorted": ConstMethodIntr(),
         "select": ConstFunctionIntr(),
         "setdiff1d": ConstFunctionIntr(),
         "shape": ConstFunctionIntr(),

--- a/pythran/tests/test_numpy_func0.py
+++ b/pythran/tests/test_numpy_func0.py
@@ -632,7 +632,7 @@ def np_rosen_der(x):
         self.run_test("def np_searchsorted1(x): from numpy import searchsorted; return searchsorted(x, 3)", numpy.arange(6), np_searchsorted1=[NDArray[int,:]])
 
     def test_searchsorted0(self):
-        self.run_test("def np_searchsorted0(x): from numpy import searchsorted; return searchsorted(x, 3, 'right')", numpy.arange(6), np_searchsorted0=[NDArray[int,:]])
+        self.run_test("def np_searchsorted0(x): from numpy import searchsorted; return x.searchsorted(3, 'right')", numpy.arange(6), np_searchsorted0=[NDArray[int,:]])
 
     def test_rot904(self):
         self.run_test("def np_rot904(x): from numpy import rot90; return rot90(x, 4)", numpy.arange(24).reshape(2,3,4), np_rot904=[NDArray[int, :, :, :]])

